dotcom-rendering/src/web/components/Picture.tsx

```diff
@@ -46,63 +48,48 @@ const getSourcesForRoleAndResolution = (
 		: setsForRole.filter((set) => !set.src.includes('dpr=2'));
 };
 
-const getFallback = (
-	role: RoleType,
-	resolution: ResolutionType,
-	imageSources: ImageSource[],
-): string | undefined => {
-	// Get the sources for this role and resolution
-	const sources: SrcSetItem[] = getSourcesForRoleAndResolution(
-		imageSources,
-		role,
-		resolution,
-	);
+const getFallback = (sources: SrcSetItem[]): string | undefined => {
 	if (sources.length === 0) return undefined;
 	// The assumption here is readers on devices that do not support srcset are likely to be on poor
 	// network connections so we're going to fallback to a small image
 	return getClosestSetForWidth(300, sources).src;
 };
 
-const getSources = (
-	role: RoleType,
-	resolution: ResolutionType,
-	imageSources: ImageSource[],
-): string => {
-	// Get the sources for this role and resolution
-	const sources: SrcSetItem[] = getSourcesForRoleAndResolution(
-		imageSources,
-		role,
-		resolution,
-	);
+// The image sources we're provided use double-widths for HDPI images
+// We should therefor multiply the width based on if it's HDPI or not
+// e.g { url: '... ?width=500&dpr=2 ...', width: '1000' }
+export const getDesiredWidth = (width: number, hdpi: boolean) =>
+	hdpi ? width * 2 : width;
 
-	return sources.map((srcSet) => `${srcSet.src} ${srcSet.width}w`).join(',');
-};
+// Turn SrsSets into a string which can be passed to the source element
+export const getSourcesFromSrcSets = (sources: SrcSetItem[]) =>
+	sources.map((srcSet) => `${srcSet.src} ${srcSet.width}w`).join(',');
 
-/**
- *       mobile: 320
- *       mobileMedium: 375
- *       mobileLandscape: 480
- *       phablet: 660
- *       tablet: 740
- *       desktop: 980
- *       leftCol: 1140
- *       wide: 1300
- */
-
-const getSizes = (
+export const optimiseBreakpointSizes = (
+	breakpointSizes: [number, Pixel][],
+): [number, Pixel][] =>
+	breakpointSizes.filter(([, size], index) => {
+		// If the NEXT element in the array has the SAME size as this one, we can trust that this current breakpoint is redundant
+		// e.g
+		// [[980, 620], [660, 620]] > [[660, 620]]
+		// Because we use min-width, any source larger than 660 will get the 620px image, even without the min-width: 980 source element
+		if (
+			index + 1 < breakpointSizes.length &&
+			breakpointSizes[index + 1][1] === size
+		)
+			return false;
+		return true;
+	});
+
+// Returns the size image desired at a particular breakpoint based on the image role, format, and if it's main media.
+const getSizeForBreakpoint = (
```

Comment:
As discussed, there's a lot of use of 'size' here but maybe what we're working with is 'width'?